test/refactoring/ConvertAsync/convert_function.swift

```diff
@@ -342,4 +363,38 @@ func withImplicitReturn(completionHandler: (String) -> Void) {
 // IMPLICIT-RETURN: func withImplicitReturn() async -> String {
 // IMPLICIT-RETURN-NEXT:   let val0 = await simple()
 // IMPLICIT-RETURN-NEXT:   return val0
-// IMPLICIT-RETURN-NEXT: }
\ No newline at end of file
+// IMPLICIT-RETURN-NEXT: }
+
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ADD-ALWAYS-NIL-ERROR %s
+func addAlwaysNilError(completion: (String?, Error?) -> Void) {
+  simple { res in
+    completion(res, nil)
+  }
+}
+// ADD-ALWAYS-NIL-ERROR:      func addAlwaysNilError() async throws -> String {
+// ADD-ALWAYS-NIL-ERROR-NEXT:   let res = await simple()
+// ADD-ALWAYS-NIL-ERROR-NEXT:   return res
+// ADD-ALWAYS-NIL-ERROR-NEXT: }
+
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=RELAY-ERROR-AND-RESULT %s
+func relayErrorAndResult(completion: (String?, Error?) -> Void) {
+  simpleErr(arg: "test") { (res, err) in
+    completion(res, err)
+  }
+}
+// RELAY-ERROR-AND-RESULT:      func relayErrorAndResult() async throws -> String {
+// RELAY-ERROR-AND-RESULT-NEXT:   let res = try await simpleErr(arg: "test")
+// RELAY-ERROR-AND-RESULT-NEXT:   return res
+// RELAY-ERROR-AND-RESULT-NEXT: }
+
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ALWAYS-THROW-ERROR-AND-RETURN-RESULT %s
+func alwaysThrowErrorAndReturnResult(completion: (String?, Error?) -> Void) {
+  simple { res in
+    completion(res, CustomError.Bad)
+  }
+}
+// ALWAYS-THROW-ERROR-AND-RETURN-RESULT:      func alwaysThrowErrorAndReturnResult() async throws -> String {
+// ALWAYS-THROW-ERROR-AND-RETURN-RESULT-NEXT:   let res = await simple()
+// ALWAYS-THROW-ERROR-AND-RETURN-RESULT-NEXT:   assert(res == nil, "The code this function was generated from passed a result and an error argument to the completion handler. Since this function just throws the error, it is assumed that the result was nil.")
+// ALWAYS-THROW-ERROR-AND-RETURN-RESULT-NEXT:   throw CustomError.Bad
+// ALWAYS-THROW-ERROR-AND-RETURN-RESULT-NEXT: }
```

Comment:
Could you make sure we have test cases for a `(String?, Error?) -> Void` completion handler where:
1. Result is non-optional and error is optional
2. Result is optional and error is non-optional
3. Both are optional
4. Both are non-optional
Ideally with simple closure args/aliases as well as unknown expressions (ie. a random call/error/whatever).

I don't see a test for (nil, nil) anywhere so if you could add that one that'd be great :).

Finally, in regards to this test specifically, I'd prefer a placeholder and comment rather than the assert (since the compiler error could be confusing for people). Suggested wording could be:
```
// The original completion handler call passed a non-optional error. An additional result cannot be represented by the async function and just a throws has been generated instead. Note that this drops the result.
```
Note that if the error expression *is* optional we could save it to a variable and then do an `if` instead (ie. the same as the continuation refactoring).

Reply:
Yeah I also think a placeholder + comment would be a little clearer to the user in that case (though I hope it doesn't come up too often)